core/src/main/java/com/google/instrumentation/trace/SpanBuilder.java

```diff
@@ -0,0 +1,277 @@
+ */
+public final class SpanBuilder {
+  private final SpanFactory spanFactory;
+  private final ContextSpanHandler contextSpanHandler;
+  private SpanContext parentSpanContext;
+  private boolean hasRemoteParent;
+  private final String name;
+  private final StartSpanOptions startSpanOption = new StartSpanOptions();
+
+  SpanBuilder(
+      SpanFactory spanFactory,
+      ContextSpanHandler contextSpanHandler,
+      SpanContext parentSpanContext,
+      boolean hasRemoteParent,
+      String name) {
+    this.parentSpanContext = parentSpanContext;
+    this.hasRemoteParent = hasRemoteParent;
+    this.name = name;
+    this.spanFactory = spanFactory;
+    this.contextSpanHandler = contextSpanHandler;
+  }
+
+  /**
+   * Sets the start time for the {@link Span}.
+   *
+   * @param startTime The start time for the {@code Span}. If {@code null} is used, then the current
+   *     system time at the point at which {@link #startSpan} is called will be used.
+   * @return this.
+   */
+  public SpanBuilder setStartTime(@Nullable Timestamp startTime) {
+    startSpanOption.setStartTime(startTime);
+    return this;
+  }
+
+  /**
+   * Sets the {@link Sampler} to use. If a {@code null} value is passed, the implementation will
+   * provide a default.
+   *
+   * @param sampler The {@code Sampler} to use when determining sampling for a {@code Span}.
+   * @return this.
+   */
+  public SpanBuilder setSampler(@Nullable Sampler sampler) {
+    startSpanOption.setSampler(sampler);
+    return this;
+  }
+
+  /**
+   * Sets the {@code List} of parent links. Links are used to link {@link Span}s in different
+   * traces. Used (for example) in batching operations, where a single batch handler processes
+   * multiple requests from different traces.
+   *
+   * @param parentLinks New links to be added.
+   * @return this.
+   */
+  public SpanBuilder setParentLinks(@Nullable List<Span> parentLinks) {
+    startSpanOption.setParentLinks(parentLinks);
+    return this;
+  }
+
+  /**
+   * Sets recordEvents.
+   *
+   * @param recordEvents New value determining if this {@code Span} should have events recorded. If
+   *     a {@code null} value is passed, the implementation will provide a default.
+   * @return this.
+   */
+  public SpanBuilder setRecordEvents(@Nullable Boolean recordEvents) {
+    startSpanOption.setRecordEvents(recordEvents);
+    return this;
+  }
+
+  /**
+   * If called this will force the newly created {@code Span} to be a root span. As a consequence,
+   * any parent specified (or inherited from the Context) will be ignored (N.B. does not apply to
+   * linked parents set through {@link #setParentLinks}).
+   *
+   * @return this.
+   */
+  public SpanBuilder becomeRoot() {
+    parentSpanContext = null;
+    hasRemoteParent = false;
+    return this;
+  }
+
+  /**
+   * Creates and starts a new {@link Span}.
+   *
+   * <p>If used without try-with-resources <b>must</b> manually call {@link Span#end}.
```

Comment:
I think this will be a less used option, as it implies manual propagation. If we remove closeable from span, it will further encourage startScoped, which more often what someone wants. For those with manual propagation, they rarely get to close in the same block as it was started.